File: subserver/pyserver/apis/_template/service.py
"""复制本目录到 apis/<组名>/ 并修改 group、routes、commands。"""

from pathlib import Path

from fastapi import Request

_PLUGIN_DIR = Path(__file__).resolve().parent


async def cmd_status(_request, _args):
    return {"service": "my-plugin", "ready": True}


async def hello_handler(_request: Request):
    return {"ok": True, "message": "hello"}


default = {
    "name": "my-plugin",
    "description": "Python 插件模板",
    "group": "my-plugin",
    "plugin_dir": str(_PLUGIN_DIR),
    "priority": 50,
    "commands": {"status": cmd_status},
    "routes": [
        {"method": "GET", "path": "/api/my-plugin/hello", "handler": hello_handler},
    ],
}

File: subserver/phpserver/apis/_template/service.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/../../core/plugin_kit.php';

$pluginDir = __DIR__;
return [
    'name' => 'my-plugin',
    'description' => '我的 PHP 插件',
    'group' => 'my-plugin',
    'pluginDir' => $pluginDir,
    'priority' => 100,
    'commands' => [
        'status' => fn () => ['service' => 'my-plugin', 'ready' => true],
    ],
    'routes' => [
        [
            'method' => 'GET',
            'path' => '/api/my-plugin/hello',
            'handler' => fn () => json_response(200, ['ok' => true, 'message' => 'hello']),
        ],
    ],
];
